app/backend/error.py

```diff
@@ -7,13 +8,22 @@
 If you are an administrator of the app, view the full error in the logs. See aka.ms/appservice-logs for more information.
 Error type: {error_type}
 """
-ERROR_MESSAGE_FILTER = """Your message contains content that was flagged by the OpenAI content filter."""
+ERROR_MESSAGE_FILTER = """Your message contains content that was flagged by the content filter."""
 
-ERROR_MESSAGE_LENGTH = """Your message exceeded the context length limit for this OpenAI model. Please shorten your message or change your settings to retrieve fewer search results."""
+ERROR_MESSAGE_LENGTH = """Your message exceeded the context length limit for this model. Please shorten your message or change your settings to retrieve fewer search results."""
+
+
+class PromptProtectionError(Exception):
+    message: str
+    code: Optional[str] = None
+
+    def __init__(self, message: str, code: Optional[str]) -> None:
+        super().__init__(message)
+        self.code = code
 
 
 def error_dict(error: Exception) -> dict:
-    if isinstance(error, APIError) and error.code == "content_filter":
+    if (isinstance(error, APIError) or isinstance(error, PromptProtectionError)) and error.code == "content_filter":
```

Comment:
Check out this cool syntactic sugar for what you're doing there (passing a tuple of types to check against)!
```suggestion
    if isinstance(error, (APIError, PromptProtectionError)) and error.code == "content_filter":
```

Reply:
Done